test/providers/summary/scancode/new-summarizer.js

```diff
@@ -58,7 +58,7 @@ describe('ScanCodeNewSummarizer basic compatability', () => {
       const coordinates = { type: 'pypi', provider: 'pypi' }
       const harvestData = getHarvestData(scancodeVersion, 'pypi-complex-declared-license')
       const result = summarizer.summarize(coordinates, harvestData)
-      assert.equal(result.licensed.declared, 'HPND')
+      assert.equal(result.licensed.declared, 'LicenseRef-scancode-secret-labs-2011')
```

Comment:
`secret-labs-2011` is the declared license according to the [raw ScanCode results](https://github.com/clearlydefined/service/blob/master/test/fixtures/scancode/32.1.0/pypi-complex-declared-license.json). Before this change, our logic fell back to the first package's declared license which is `HPND`.

I'm not sure which is the ultimately correct one but we need this change to surface the ScanCode result.

Reply:
I want to be sure I am understanding `logic fell back to the first package's declared license` correctly.  Looking at the [fixture](https://raw.githubusercontent.com/clearlydefined/service/ed446642d56cf0698098f9c147d2702ad2e6141b/test/fixtures/scancode/32.1.0/pypi-complex-declared-license.json), I see a license under summary which seems like the correct license...

```
    "summary": {
      "declared_license_expression": "secret-labs-2011",
```

and farther down, I see the first package (transient dependency) has the `HPND` as its declared license...

```
    "packages": [
      {
        "type": "pypi",
        "namespace": null,
        "name": "Pillow",
        "version": "9.5.0",
        ...
        "declared_license_expression": "historical",
        "declared_license_expression_spdx": "HPND",
```

It would be interesting to understand if that is a correct interpretation of how `HPND` was identified as the license and why that approach was chosen.  To me, that doesn't seem correct as that is the license for Pillow 9.5.0.

@qtomlinson any insights into this?

Reply:
In [v30 result](https://[raw.githubusercontent.com/clearlydefined/service/master/test/fixtures/scancode/30.1.0/pypi-complex-declared-license.json) (line 760- 766) shows content.packages[0].declared_license as HPND
```
          "license_expression": "historical",
          "declared_license": {
            "license": "HPND",
            "classifiers": [
              "License :: OSI Approved :: Historical Permission Notice and Disclaimer (HPND)"
            ]
          },
```

Reply:
In [v30 result](https://[raw.githubusercontent.com/clearlydefined/service/master/test/fixtures/scancode/30.1.0/pypi-complex-declared-license.json) (line 760-766) shows content.packages[0].declared_license as HPND
          "license_expression": "historical",
          "declared_license": {
            "license": "HPND",
            "classifiers": [
              "License :: OSI Approved :: Historical Permission Notice and Disclaimer (HPND)"
            ]
          },

Reply:
[Pypi](https://pypi.org/project/pillow/9.5.0) also shows "License as [OSI Approved :: Historical Permission Notice and Disclaimer (HPND)]"

Reply:
@yashkohli88, Could you please provide some feedback regarding the license of pypi Pillow 9.5.0?

Reply:
> `secret-labs-2011` is the declared license according to the raw [ScanCode results](https://github.com/clearlydefined/service/blob/master/test/fixtures/scancode/32.1.0/pypi-complex-declared-license.json). 

Just noticed another case where declared_license_expression (v32) seems to be different from what is declared from the package.  Added here for documentation purposes.
[32.3.0.json](https://github.com/user-attachments/files/16184402/32.3.0.json):
```
    "summary": {
      "declared_license_expression": "cc-by-4.0 AND cc-by-sa-4.0 AND gpl-2.0",
	  ...
    package[0]
        "declared_license_expression": "gpl-2.0-plus AND gpl-2.0",
        "declared_license_expression_spdx": "GPL-2.0-or-later AND GPL-2.0-only",
	...
    files:
    {
        "path": "pylint-3.2.3/LICENSE",
        "detected_license_expression": "gpl-2.0",
        "detected_license_expression_spdx": "GPL-2.0-only",
```
[30.3.0.json](https://github.com/user-attachments/files/16184404/30.3.0.json):
```
    package[0]					
	"license_expression": "gpl-2.0-plus AND gpl-2.0",
	"declared_license": {
		"license": "GPL-2.0-or-later",
		"classifiers": [
			"License :: OSI Approved :: GNU General Public License v2 (GPLv2)"
		]
	},
	....
	files: 
	{
		"path": "pylint-3.2.3/LICENSE",
				"key": "gpl-2.0",
```
"cc-by-4.0 AND cc-by-sa-4.0 AND gpl-2.0" in v32 is different from "gpl-2.0-plus AND gpl-2.0" in v30

Reply:
Either way, I think all of this cases are bugs/regressions in ScanCode, right? Meaning, our code is behaving as expected here, just producing unexpected/wrong results based on the underlying raw data 🤔